osu.Game/Screens/Edit/Compose/Components/PositionSnapGrid.cs

```diff
@@ -49,10 +50,17 @@ protected override void Update()
 
         protected abstract void CreateContent();
 
+        /// <summary>
+        /// Computes the line thickness of this <see cref="PositionSnapGrid"/> independent of display resolution.
+        /// </summary>
+        /// <remarks>
+        /// Unless resolution is high enough, in which case thickness will remain the same.
+        /// </remarks>
+        protected float GetLineWidth() => Math.Max(DrawWidth / ScreenSpaceDrawQuad.Width, 0.4f);
```

Comment:
On the master screenshot, the lines are all the same size, on the PR, the lines have varying thickness which looks bad (at least when zooming the image to native resolution on 1080p screen).

Maybe having the lines always be an integer pixel size would fix this issue. 1px on 1080p, scaling up to 2px, 3px, etc. when necessary.

First check if 2px size looks good at 4K!

Reply:
Indeed looks better
|master|(pr + adjustment)|
|---|---|
|![master](https://github.com/user-attachments/assets/a3ca7423-9700-4c67-bac2-ec51792f9fb1)|![pr](https://github.com/user-attachments/assets/4be5e776-fc93-49e6-912d-04fa7f230b7d)|